modules/redhen_connection/src/Plugin/Action/ActivateConnection.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Drupal\redhen_connection\Plugin\Action;
+
+use Drupal\Core\Action\ActionBase;
+use Drupal\Core\Session\AccountInterface;
+
+/**
+ * Activates a Connection.
+ *
+ * @Action(
+ *   id = "connection_activate_connection_action",
+ *   label = @Translation("Activate the selected connection(s)"),
+ *   type = "redhen_connection"
+ * )
+ */
+class ActivateConnection extends ActionBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute($connection = NULL) {
+    // Skip activating the connection already active.
+    if ($connection !== FALSE && !$connection->isActive()) {
+      // For efficiency manually save the original account before applying any
```

Comment:
```suggestion
      // For efficiency manually save the original connection before applying any
```
I don't really understand what this step is doing for us. Why do we need an `->original` at all, and why would we be setting it in the middle of a VBO action? It seems like a lot of activity for something that should be able to execute quickly on a large number of entities.

Reply:
I borrowed a lot of this from the core User module, not sure why I kept that in there. 🤔 
Removed.